driver-core/src/main/java/com/datastax/driver/core/ScyllaCloudDatacenter.java

```diff
@@ -0,0 +1,113 @@
+      @JsonProperty(value = "proxyURL") String proxyURL) {
+    this.certificateAuthorityPath = certificateAuthorityPath;
+    this.certificateAuthorityData = certificateAuthorityData;
+    this.server = server;
+    this.TLSServerName = TLSServerName;
+    this.nodeDomain = nodeDomain;
+    this.proxyURL = proxyURL;
+  }
+
+  public void validate() {
+    if (certificateAuthorityData == null) {
+      if (certificateAuthorityPath == null) {
+        throw new IllegalArgumentException(
+            "Either certificateAuthorityData or certificateAuthorityPath must be provided for datacenter description.");
+      }
+      File file = new File(certificateAuthorityPath);
+      if (!file.canRead()) {
+        throw new IllegalArgumentException(
+            "Cannot read file at given certificateAuthorityPath ("
+                + certificateAuthorityPath
+                + ").");
+      }
+    }
+    if (server == null) {
+      throw new IllegalArgumentException("server property is required in datacenter description.");
+    } else {
+      try {
+        // Property 'server' is not a true URL because it does not contain protocol prefix
+        // We're adding prefix just to satisfy that part of validation
+        URL url = new URL("http://" + server);
+        if (url.getPort() == -1) {
+          throw new IllegalArgumentException(
+              "server property '" + server + "' does not contain a port.");
+        }
+      } catch (MalformedURLException e) {
+        throw new IllegalArgumentException(
+            "server property '" + server + "' is not a valid URL", e);
+      }
+    }
+    if (nodeDomain == null || nodeDomain.length() == 0) {
+      throw new IllegalArgumentException(
+          "nodeDomain property is required in datacenter description.");
+    } else {
+      if (nodeDomain.contains(" "))
+        throw new IllegalArgumentException(
+            "nodeDomain '" + nodeDomain + "' cannot contain spaces.");
+      if (nodeDomain.startsWith(".") || nodeDomain.endsWith("."))
+        throw new IllegalArgumentException(
+            "nodeDomain '" + nodeDomain + "' cannot start or end with a dot.");
+      try {
+        // Lenient validation
+        InternetDomainName.isValid(nodeDomain);
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            "nodeDomain '" + nodeDomain + "' is not a valid domain name.", e);
+      }
+    }
+  }
+
+  public String getCertificateAuthorityPath() {
+    return certificateAuthorityPath;
+  }
+
+  public byte[] getCertificateAuthorityData() {
+    return certificateAuthorityData;
+  }
+
+  public InetSocketAddress getServer() {
+    HostAndPort parsedServer = HostAndPort.fromString(server);
+    return InetSocketAddress.createUnresolved(parsedServer.getHostText(), parsedServer.getPort());
+  }
+
+  public String getNodeDomain() {
+    return nodeDomain;
+  }
+
+  public String getTLSServerName() {
+    return TLSServerName;
```

Comment:
this field start with upper case letter contrary to how other fields are called. I understand it's a abbreviation, but for SNI this uppercase isn't used. Would be good to be consistent.